stockflux-securities-master/src/components/InputForm.js

```diff
@@ -1,25 +1,54 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
+import Components from "stockflux-components";
 import { Link } from "react-router-dom";
 import "./InputForm.css";
+import { getSecurity } from "../services/SecuritiesService";
+import ErrorMessage from "./ErrorMessage";
 
 const InputForm = ({ match }) => {
   const [name, setName] = useState("");
   const [exchange, setExchange] = useState("");
   const [symbol, setSymbol] = useState("");
-  const [isShown, setIsShown] = useState(false);
-  const [isEnabled, setIsEnabled] = useState(false);
+  const [visible, setVisible] = useState(false);
+  const [enabled, setEnabled] = useState(false);
   const [inProgress, setInProgress] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState(null);
+
+  const setSecurityState = security => {
+    setName(security.name);
+    setExchange(security.exchange);
+    setSymbol(security.symbol);
+    setVisible(security.visible);
+    setEnabled(security.enabled);
+  };
+
+  useEffect(() => {
+    if (match.params.securityId) {
+      setIsLoading(true);
+      getSecurity(match.params.securityId)
+        .then(security => {
+          setIsLoading(false);
+          setErrorMessage(null);
+          setSecurityState(security);
+        })
+        .catch(() => {
+          setIsLoading(false);
+          setErrorMessage("Error, cannot get security");
+        });
+    }
+  }, []);
 
   const submitForm = event => {
     event.preventDefault();
     setInProgress(true);
 
     const securityObject = {
```

Comment:
Just a tip regarding ES6 here. If the name of the property in an object matches the name of the variable, you can simply omit the value and it will have the same effect. For example, this object could become:

    const securityObject = { name, exchange, symbol, visible, enabled };

Reply:
yeah good spot there, changed now